python/ray/data/block.py

```diff
@@ -213,7 +213,7 @@ class BlockMetadata(BlockStats):
 
     def to_stats(self):
         return BlockStats(
-            **{k: v for k, v in asdict(self).items() if k in _BLOCK_STATS_FIELD_NAMES}
+            **{key: self.__getattribute__(key) for key in _BLOCK_STATS_FIELD_NAMES}
```

Comment:
not sure how much faster this is gonna be. 
Can you do a simple microbenchmark?
If it's still non trivial. We'll need to move `to_stats` out of the critical path. Currently called  [here](https://github.com/ray-project/ray/blob/ff9f0bc8010cfd2e7759a64684f77192d644e6f7/python/ray/data/_internal/execution/operators/map_operator.py#L460-L460).

Reply:
Yes, this is substantially faster as this doesn't do any deep-copying just doing the lookups

Reply:
Alright, so doing 1M `to_stats`

Before:

```
(ray-312) ➜  ray git:(ak/blk-stt-fix) ✗ python test_blk_mdata.py
>>> Took: 23.48s
```

After:

```
(ray-312) ➜  ray git:(ak/blk-stt-fix) ✗ python test_blk_mdata.py
>>> Took: 0.52s
```